frontend/packages/console-dynamic-plugin-sdk/src/app/components/utils/rbac.tsx

```diff
@@ -0,0 +1,150 @@
+};
+
+/**
+ * Provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Object with resource access information.
+ * * */
+export const checkAccess = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?: ImpersonateKind,
+): Promise<SelfSubjectAccessReviewKind> => {
+  // Destructure the attributes with defaults so we can create a stable cache key.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes || {};
+  return checkAccessInternal(
+    group,
+    resource,
+    subresource,
+    verb,
+    name,
+    namespace,
+    getImpersonateKey(impersonate),
+  );
+};
+
+/**
+ * Hook that provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return Array with isAllowed and loading values.
+ * * */
+export const useAccessReview2 = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?: ImpersonateKind,
+): [boolean, boolean] => {
+  const [loading, setLoading] = useSafetyFirst(true);
+  const [isAllowed, setAllowed] = useSafetyFirst(false);
+  // Destructure the attributes to pass them as dependencies to `useEffect`,
+  // which doesn't do deep comparison of object dependencies.
+  const {
+    group = '',
+    resource = '',
+    subresource = '',
+    verb = '' as K8sVerb,
+    name = '',
+    namespace = '',
+  } = resourceAttributes;
+  const impersonateKey = getImpersonateKey(impersonate);
+  React.useEffect(() => {
+    checkAccessInternal(group, resource, subresource, verb, name, namespace, impersonateKey)
+      .then((result: SelfSubjectAccessReviewKind) => {
+        setAllowed(result.status.allowed);
+        setLoading(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('SelfSubjectAccessReview failed', e);
+        // Default to enabling the action if the access review fails so that we
+        // don't incorrectly block users from actions they can perform. The server
+        // still enforces access control.
+        setAllowed(true);
+        setLoading(false);
+      });
+  }, [setLoading, setAllowed, group, resource, subresource, verb, name, namespace, impersonateKey]);
+
+  return [isAllowed, loading];
+};
+
+/**
+ * Hook that provides information about user access to a given resource.
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @return The isAllowed boolean value.
+ * * */
+export const useAccessReview = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?: ImpersonateKind,
+): boolean => {
+  return useAccessReview2(resourceAttributes, impersonate)[0];
+};
```

Comment:
Okay, we need to rename these -- exporting this "1" "2" thing is not a good look for the SDK.

I think they do have different uses and we can support that, but we need to rename them -- which is annoying for the imports since you went directly to each file 😞 

Here is what I am thinking:

- `useAccessReview` is something like `useAccessReviewAllowed`
    - Update the comment string to say it gives access; maybe something like:
        > Hook that provides allowed status about user access to a given resource
- `useAccessReview2` becomes `useAccessReview` -- since it's the base entry and this entry is just wrapping for a specific use-case

Reply:
@andrewballantyne renamed the exports in the SDK, do we want to also rename these globally across the console? It's around 100 files and I can do it, just would like to confirm with you before. At the same time, I could replace the imports to point to our new SDK location of these hooks.